modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java

```diff
@@ -1455,6 +1493,9 @@ void undeploy(@NotNull IgniteUuid srvcId) {
 
         if (ctxs != null) {
             synchronized (ctxs) {
+                if (!ctxs.isEmpty())
+                    ctx.metric().remove(serviceMetricRegistryName(ctxs.iterator().next().name()));
```

Comment:
Should we made this a part of the `cancel` method, before looping over service contexts? Currently you removing registry twice in your code, and both times before cancellation of services.

Reply:
Fixed